### PR TITLE
test: testing the return of func constructAuthorizationModelFromSQLRows

### PR DESCRIPTION
**Pull request by Madraceee**
## Description
Added tests to check the return value of function constructAuthorizationModelFromSQLRows

## References
Issue #1952

## Review Checklist
- [ ] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [ ] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: pkg/storage/mysql/mysql_test.go [hunk 1]
```diff
@@ -293,6 +293,32 @@ func TestReadAuthorizationModelUnmarshallError(t *testing.T) {
...
+
+	_, err = ds.db.ExecContext(ctx, "INSERT INTO authorization_model (store, authorization_model_id, schema_version, type, type_definition, serialized_protobuf) VALUES (?, ?, ?, ?, ?, ?)", store, modelID, schemaVersion, "document", bytes, nil)
+
+	require.NoError(t, err)
+
+	res, err := ds.ReadAuthorizationModel(ctx, store, modelID)
+	require.NoError(t, err)
+	// AuthorizationModel should return only 1 type which is of type "document"
+	require.Equal(t, 1, len(res.TypeDefinitions))
+	require.Equal(t, "document", res.TypeDefinitions[0].Type)
```
Comment (adriantam) [inline pkg/storage/mysql/mysql_test.go:320]:
```suggestion
	require.Len(t, res.GetTypeDefinitions(), 1)
	require.Equal(t, "document", res.GetTypeDefinitions()[0].Type)
```

File: pkg/storage/postgres/postgres_test.go [hunk 2]
```diff
@@ -291,6 +291,34 @@ func TestReadAuthorizationModelUnmarshallError(t *testing.T) {
...
+	_, err = ds.db.ExecContext(ctx, "INSERT INTO authorization_model (store, authorization_model_id, schema_version, type, type_definition, serialized_protobuf) VALUES ($1, $2, $3, $4, $5, $6)", store, modelID, schemaVersion, "document", bytes, nil)
+
+	require.NoError(t, err)
+
+	res, err := ds.ReadAuthorizationModel(ctx, store, modelID)
+
+	require.NoError(t, err)
+	// AuthorizationModel should return only 1 type which is of type "document"
+	require.Equal(t, 1, len(res.TypeDefinitions))
+	require.Equal(t, "document", res.TypeDefinitions[0].Type)
```
Comment (adriantam) [inline pkg/storage/postgres/postgres_test.go:319]:
```suggestion
	require.Len(t, res.GetTypeDefinitions(), 1)
	require.Equal(t, "document", res.GetTypeDefinitions()[0].Type)
```

File: pkg/storage/mysql/mysql_test.go (hunk 1 above)
Comment (Madraceee) [reply pkg/storage/mysql/mysql_test.go]: Updated Test file

File: pkg/storage/postgres/postgres_test.go (hunk 2 above)
Comment (Madraceee) [reply pkg/storage/postgres/postgres_test.go]: Updated Test file

**Comment by Madraceee** [comment]
Can any of the maintaines help me as to why the tests  are failing?
Is it because its behind the main branch or something else?
All the tests are passing locally

**Comment by adriantam** [comment]
> Can any of the maintaines help me as to why the tests are failing? Is it because its behind the main branch or something else? All the tests are passing locally

Unfortunately, the test infrastructure is flaky for MySQL.

**Review by miparnisari — APPROVED** [review]

**Pull request merged**